pages/api/situation.js

```diff
@@ -1,11 +1,22 @@
 import cookie from "cookie";
 
 export default (req, res) => {
+  const existingSituation = JSON.parse(req.cookies?.situation ?? "{}");
+
+  // update the user situation cookie
   if (req.method === "POST") {
-    // update the user situation cookie
+    const updatedSituation = req.body.situation || {};
+    const situation = {
+      location: updatedSituation.location
+        ? updatedSituation.location
+        : existingSituation.location ?? "",
+      age: updatedSituation.age || (existingSituation.age ?? ""),
+      income: updatedSituation.income || (existingSituation.income ?? ""),
+    };
```

Comment:
We should assume the situation being passed is always current and correct. If someone removed a value from their situation we wouldn't want to hold onto existing information